test/scenarios.py

```diff
@@ -5947,4 +5947,116 @@
+                                },
+                            ),
+                            event.FinishFailedTransfer(
+                                2,
+                                Error.IO,
+                            ),
+                        ]
+                    ),
+                    action.Sleep(2),
+                    action.NewTransfer("172.20.0.15", ["/tmp/testfile-small"]),
+                    action.Wait(
+                        event.Queued(
+                            3,
+                            {
+                                event.File(
+                                    FILES["testfile-small"].id,
+                                    "testfile-small",
+                                    1048576,
+                                ),
+                            },
+                        )
+                    ),
+                    action.ExpectCancel([0, 1, 3], True),
+                    action.Stop(),
+                ]
+            ),
+            "stimpy": ActionList(
+                [
+                    # There are 2 request per transfer, the first one for
+                    # authentication nonce and the second with transfer. One
+                    # more is needed because that's how the implementation
+                    # works.
+                    action.Start("172.20.0.15", max_reqs=5),
+                    action.WaitRacy(
+                        [
+                            event.Receive(
+                                0,
+                                "172.20.0.5",
+                                {
+                                    event.File(
+                                        FILES["testfile-small"].id,
+                                        "testfile-small",
+                                        1048576,
+                                    ),
+                                },
+                            ),
+                            event.Receive(
+                                1,
+                                "172.20.0.5",
+                                {
+                                    event.File(
+                                        FILES["testfile-small"].id,
+                                        "testfile-small",
+                                        1048576,
+                                    ),
+                                },
+                            ),
+                        ]
+                    ),
+                    action.Wait(
+                        event.Receive(
+                            2,
+                            "172.20.0.5",
+                            {
+                                event.File(
+                                    FILES["testfile-small"].id,
+                                    "testfile-small",
+                                    1048576,
+                                ),
+                            },
+                        )
+                    ),
```

Comment:
I think it might help to use different files in transfers to better distinguish the transfers, however, what's weird is that the receiver is made aware of the transfer via `RequestReceived` but then is not made aware of it's failure so we should either send an additional failure event as well or don't emit the event at all.

I would prefer not emitting any event at all in this case as we can flood the receiver with events as a DDoS

Reply:
The receiver does not get any event in case of failure. I think you might have misunderstood this as the UUID slots are local to the runner so Stimpy's slot 2 is the same transfer as Ren's slot 3 in this case

Reply:
I mean receiver receives the `RequestReceived` then it responds with TooManyRequests behind the scenes to the sender but the receiver itself is oblivious that the transfer is failed

Reply:
True, that's what happens here. Do you suggest something else? I would argue that the current approach is correct

Reply:
Ok so from the beginning. If we are a receiver and a peer makes HTTP request to us and the DDoS protection is triggered we do not receive any event. As a user, we basically don't know about the request at all.
The sender however receives a `TransferFailed` event.

Is this approach somehow wrong in your opinion?

Reply:
That sounds very correct and exactly how I would expect it to be. It looks correct now as it actually initiates with 3 transfers instead of 2(i looked only at events), sorry about the confusion.